Iss404

kmartinchek opened the pull request:
**Living Wage Ratio**

Please include the following points in your PR:

1) A link to the issue that this PR relates to. #67  

2) A description of the content in this pull request.  
- What was changed? In this update, I created new Stata code to generate the living wage ratio from 2015 to 2022, both overall and by subgroup. This will be able to complete the backfill for this calculation on all prior years in two do files, instead of one file per year. 
- What should the reviewer be focusing on? [compute_metrics_wage_ratio_overall.do](https://github.com/UI-Research/mobility-from-poverty/blob/iss404/09_employment/compute_metrics_wage_ratio_overall.do) and [compute_metrics_wage_ratio_subgroup.do](https://github.com/UI-Research/mobility-from-poverty/blob/iss404/09_employment/compute_metrics_wage_ratio_subgroup.do) . These add new years and subgroup data for 2015 to 2022-- checking data is as expected, loops work properly, etc. would be excellent. I make manual adjustments for geography incompatibilities in 2022. We deflate/inflate MIT data for 2015 to 2022 due to data availability constraints and 2023 data will be added once it is available.
- Is there a logical order to review the files in? Overall then subgroup file.

3) Detail on any issues or flags that the metric reviewer/data-team should be aware of. None I am aware of. I plan to combine the overall and subgroup files possibly into one do file for efficiency, but that is a matter of putting the two scripts into one script.

tinatinc commented on 09_employment/compute_metrics_wage_ratio_overall.do:
```diff
@@ -0,0 +1,252 @@
+
+
+/***** update these directories *****/
+global raw "C:\Users\KMartinchek\Documents\upward-mobility-2025\mobility-from-poverty\09_employment"
+global wages "C:\Users\KMartinchek\Documents\upward-mobility-2025\mobility-from-poverty\09_employment"
```
For this whole section, I think you should add in some instructions for anyone who will try to run this on their own computer. e.g., replace these pathways with your own local paths, etc.

tinatinc commented on 09_employment/compute_metrics_wage_ratio_overall.do:
```diff
@@ -0,0 +1,252 @@
+
+The data from MIT and QECW cannot be easily read directly into this program. 
+
+Before running, please download the files below from the following [Box folder] https://urbanorg.app.box.com/folder/298586735341 into the repository folder  
+
```
Be specific - which folder do you want them importing it into? Do they have to create the folder? Is that folder included in the gitignore?

tinatinc commented on 09_employment/compute_metrics_wage_ratio_overall.do:
```diff
@@ -0,0 +1,252 @@
+
+/***** import QCEW data *****/
+
+foreach year in 2015 2016 2017 2018 2019 2020 2021 2022 {
+	
+	clear
+	cd `raw'
+	import delimited using "`year'_data.csv", numericcols(14 15 16 17 18) varnames(1) clear
+
+	/* keep only county totals */
+	keep if areatype == "County" & ownership == "Total Covered"
+
+	keep st cnty annualaverageweeklywage annualaverageestablishmentcount
+
+	rename st state
+
+	rename cnty county
+
+	destring state, replace // make numeric for merging with MIT data
+	
+	save "temp-qecw-`year'.dta", replace 
+	
+}
```
I'm getting an error that 2020_data.csv not found

tinatinc commented on 09_employment/compute_metrics_wage_ratio_overall.do:
```diff
@@ -0,0 +1,252 @@
+	use "temp-merged-`year'.dta", clear
+
+	/* convert living hourly wage to weekly */
+	gen weekly_living_wage = wage_adj * 40
```
be specific for general audience - 40 because 40 working hours per week

tinatinc commented on 09_employment/compute_metrics_wage_ratio_subgroup.do:
```diff
@@ -0,0 +1,370 @@
+	// save out for merge
+	save "$wages/temp-qecw-industry-`year'.dta", replace
+
+}
```
same issue here - 2020_data.csv not found

tinatinc commented on 09_employment/compute_metrics_wage_ratio_subgroup.do:
```diff
@@ -0,0 +1,370 @@
+
+	*** assert number of counties
+	di `year'
+	distinct state county, joint // to confirm there are the right number of counties in the dataset
```
Cool - I didn't know this command/package existed

tinatinc commented:
Hey Kassie - this looks very strong! But I have not been able to do a thorough review beyond these comments. Could you check on this missing data file and let me know if I'm missing something before I re-run?

kmartinchek commented:
**Living Wage Ratio** 
**UPDATED DETAILS FOR MOST RECENT UPDATE**

Please include the following points in your PR:

A link to the issue that this PR relates to. https://github.com/UI-Research/mobility-from-poverty/pull/67

A description of the content in this pull request.

What was changed? In this update, I created new Stata code to generate the living wage ratio for 2014, 2018, 2021, 2022 and 2023 pursuant to our data use agreement, both overall and by subgroup. This will be able to complete the backfill for this calculation on all prior years in one do file, instead of several scripts per year in the original code.

What should the reviewer be focusing on? [compute_metrics_wage_ratio.do](https://github.com/UI-Research/mobility-from-poverty/blob/iss404/09_employment/compute_metrics_wage_ratio.do). These add new years and subgroup data for 2014, 2018, 2021, 2022, and 2023-- checking data is as expected, loops work properly, etc. would be excellent. I make manual adjustments for geography incompatibilities in 2022 and 2023 to align with guidance on CT geographies. We deflate/inflate MIT data for 2014, 2018, and 2021 pursuant to our data use agreement. Question of whether we will make publicly available the MIT files for 2019, 2022, and 2023 we use to create the measure and the best place to store and read in these files to best align with the data use agreement-- this is a question for Claudia specifically.

Is there a logical order to review the files in? Only file to review.

Detail on any issues or flags that the metric reviewer/data-team should be aware of. None I am aware of.

cdsolari commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+
+/* delete unneeded files -- do this as a last step */
+
+foreach year in 2014 2018 2021 2022 2023 {
```
I assume you haven't run this yet since this isn't finalized. We definitely need to clean up some of these extra files. If it helps, you can save some in the team Box folder, given that some of the data online might go away. But, please do remove this from github at the end.

rpitingolo replied:
@cdsolari agree. I don't know if there is enough time left to do a good spring cleaning, but ideally there would be a 'temp' folder inside of 09 and then that folder would be in the .gitignore so that the files never get pushed. If there's not enough time then we should just delete any temp datasets so they don't get pushed.

kmartinchek replied:
I do remove these files but didn't push the non-do-file changes in this review. I will do that now.

cdsolari commented:
Oh, @rpitingolo and @kmartinchek I remembered that I had not yet addressed this question: "Question of whether we will make publicly available the MIT files for 2019, 2022, and 2023 we use to create the measure and the best place to store and read in these files to best align with the data use agreement." We need to strip the MIT raw files for 2019, 2022, and 2023 from our github. This is technically public, and that violates our agreement. We should store these in our Box folder in the "Metrics_2025_round/living_wage" folder on Box. We also need to remove any of the past data from "scrape" efforts. I can, however, coordinate with @jwalsh28 to remove those. I know in R we use git ignore, and I noticed you use "erase" in STATA. I already have the MIT zip file they delivered to us in that Box folder. Let me know if you have other questions!

rpitingolo commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+			replace wage_adj = wage *  251.107/255.657 if year == 2018
+			replace wage_adj = wage *  236.736/255.657 if year == 2014
+
+			replace wage_adj = wage *  270.970/255.657 if year == 2021
```
@cdsolari I believe the other years are imported in separate loops as the structure of the data is slightly different.

My question is why is 2019 used as the base year for inflation? Maybe there is a good reason or maybe it's because it's what was used before and should be updated. 

We should create a global variable with the base year value (denominator) to make sure it is applied consistently.

cdsolari replied:
@rpitingolo - 2019 is used because that is the year of data we had scraped from the MIT Living Wage website. That's really it. Nothing otherwise intentional.

rpitingolo commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+bysort year: count
+
+// export final dataset
+export delimited using "$wages/living_wage_county_industry_longitudinal.csv", replace	
```
@cdsolari I agree. /data/final/ seems like a good place to save them.

kmartinchek replied:
I created new files here and didn't push them because I thought reviewers would be replicating locally. I will commit them now.

rpitingolo replied:
Thanks!

rpitingolo commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+	** add in county name and state name
+	import delimited using "$crosswalk/county-populations.csv", stringcols(2 3 4 5) varnames(1) clear
+
+	merge 1:m year state county using "$wages/wage_ratio_final_`year'_subgroup.dta"
```
Should non-matches be dropped here? It looks like there is code later in the loop to only keep data for the current year so it may accomplish the same goal but wanted to double check something wasn't missed here.

kmartinchek replied:
This later code accomplishes addressing the mismatches due to counties not present in the current year(s) of data.

rpitingolo commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+	replace subgroup = 1 if subgroup == .
+	label values subgroup industry_labels_mm
+
+	fillin fips_code subgroup
```
What exactly does `fillin` do? I'm not familiar with this syntax and after reading the documentation still don't fully understand what is supposed to be happening here. A more detailed comment would be helpful.

kmartinchek replied:
fillin ensure that there is a row for each industry for each county in the dataset. Some counties do not have certain industries present in the original data, so we need to add these rows and then replace the metric and quality variable values with "NA"s.

rpitingolo replied:
Interesting. Thanks.

rpitingolo commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+
+	*** assert number of counties
+	di `year'
+	distinct state county, joint // to confirm there are the right number of counties in the dataset
```
`distinct` requires installation of a package, otherwise it will error if someone is trying to run it without having the package. Please add the installation code to the top of the script.

kmartinchek replied:
Added.

rpitingolo replied:
Thank you

rpitingolo commented:
> I know in R we use git ignore, and I noticed you use "erase" in STATA. I already have the MIT zip file they delivered to us in that Box folder. Let me know if you have other questions!

Gitignore can be used for any kinds of files or folders (not just R) and I would strongly suggest using it here too!

cdsolari commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+	/* only keep 1 adult, 2 children row */
+	keep if adults == "1 Adult" & children == "2 Children"
+	
+		/* inflate/deflate MIT data, depending on year */
```
@rpitingolo Yes, that's because we have valid files from MIT Living Wage for those 3 years - 2019, 2022, and 2023 only. We will now have to pay for any other living wage data we get from them. My thing was that I was under the impression that we used 2022 living wage data and deflated it for the 2021 metric. However, it seems like this code is using 2019. Given that 2021 and 2022 are both post-pandemic, I would feel better about the 2021 estimates being based on the 2022 living wage figures, but @kmartinchek if you and Greg discussed something else and you altered that on purpose, please explain in the code and that's fine. I just want to make sure all of our table notes end up reflecting the reality. Right now, we have table notes for the 2021 data saying they are based on deflated 2022 living wage numbers. THANKS!

kmartinchek replied:
I have moved the 2021 data to the 2022 loop, as noted above. I leave the comment as-is re: deflation/inflation because this applies now to 2021 data year now implemented in this loop. 

In terms of real dollar values, I have adjusted wages in accordance with previously published UMF data, which you can see in earlier do files. As such, I haven't changed these. Please let me know if this is not appropriate.

kmartinchek commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+	keep if areatype == "County" 
+
+	** recode industries into MM categories
+	generate industry_type = .
```
It does. Line 157 confirms that super-sections and unclassified NAICS codes are unassigned to an industry type so we don't double-count.

kmartinchek commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+
+The data from MIT and QECW cannot be easily read directly into this program. 
+
+Before running, please download the files below from the following [Box folder] https://urbanorg.app.box.com/folder/298586735341 into the repository folder  
```
This link would need to change for each year of data and would be inefficient for importing QECW data. I recommend keeping as-is.

rpitingolo replied:
OK. In the interest of time I think we should keep it as-is. If we do this again in the future I think we should try to automate this part. If we don't use the sample code to import individual files we could at least put them in a .zip file on Box, download that, use Stata to extract it, then import.

kmartinchek commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+foreach year in 2014 2018 2021 {
+	
+	clear
+	cd `wages'
```
Removed.

rpitingolo replied:
Thanks

kmartinchek commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+
+	** recode industries into MM categories
+	generate industry_type = .
+		replace industry_type = 1 if naics == 10 & own == 0 // all
```
Added a link for the codes for this variable as definition.

kmartinchek commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+	drop county
+	rename new_county county
+
+	gen new_ratio = string(ratio_living_wage)
```
Yes.

kmartinchek commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+
+	*** fill in needed observations so each county has a row for each industry subgroup
+	*** including gernating all needed variables for the newly generated rows
+	egen fips_code = concat(state county)
```
Understood, but under time constraints, since this is not an error, I will not re-organize the file in this update.

kmartinchek commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+	gen ratio_living_wage = annualaverageweeklywage/weekly_living_wage
+
+	/* create data quality flag 
+	per discussion with Greg, >= 30 is 1, <30 is 3 */
```
This is NA (see line 288) -- can't be calculated because no annual establishment count.

kmartinchek commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+	replace ratio_living_wage_quality = . if avgestct == .
+	replace ratio_living_wage_quality = . if statuscode == 1
+
+	/* test flag */
```
To tab the quality variable, as indicated in line 354 which follows. I have changed the comment wording to be clearer.

kmartinchek commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+	/* test flag */
+	tab ratio_living_wage_quality, missing
+
+	/* put state and county in string with leading 0s */
```
See comment above, as this refers to same lines of code.

kmartinchek commented:
> @kmartinchek Also, just to add, that in looking at the output files Rob saved for me to review in Box, I did notice that your overall file has cases where the data quality is NA, but there is a value for the metric. This violates our rule, and it looks like it worked successfully in the industry subgroup output file, but not the overall. I just want to make sure that is resolved before this is finalized.
> 
> Also, I did notice quite a few more NA values in 2023 compared to earlier years. I am curious why that might be. I wonder how the MIT export data file we are using for 2023, which was generated in a different way than the scraped versions, looks compared to those other versions, and if we might have something to approach MIT about.
> 
> Along with Rob's comment to use gitignore, which would be my preference, I wonder if you could save your final output files in the data/final folder here: https://github.com/UI-Research/mobility-from-poverty/tree/iss404/09_employment/data/final We currently have so many output files in the 09_employment folder that I don't want to risk reading in the wrong csv file intended to be the final data file.
> 
> I'm happy to discuss any of the comments. THANKS!

I don't think the files saved there match what I generated. I have an assert statement that quality == NA when metric == NA at multiple points in the script and it does not fail. I added another at the end as well-- and I get the same result.

As mentioned in my earlier comments, I have ensured that the final output files are directed there.

I have addressed the missingness in 2023, which should now be similar scale as prior years. No need to contact MIT-- it was a QECW formatting change.

kmartinchek commented on 09_employment/compute_metrics_wage_ratio.do:
```diff
@@ -0,0 +1,559 @@
+
+	destring state, replace // make numeric for merging with MIT data
+	
+	save "temp-qecw-`year'.dta", replace 
```
I have addressed this. See comment to Claudia at end.

rpitingolo replied:
I'm not sure I am following the comment at the end. I see you added an mdesc command which is helpful but just confirms the same missingness I noticed earlier. 

If this is not actually an issue then we can resolve this one.

rpitingolo replied:
Also `mdesc` is a package that needs to be installed so please make sure the installation code is included at the top of the script.

kmartinchek replied:
I do not see the missingness you see. I recommend re-downloading the source data from the Box folder (I re-added these files just now as extra insurance). I added the code re: ssc install at top.

This should resolve the issue.

kmartinchek commented:
> > I know in R we use git ignore, and I noticed you use "erase" in STATA. I already have the MIT zip file they delivered to us in that Box folder. Let me know if you have other questions!
> 
> Gitignore can be used for any kinds of files or folders (not just R) and I would strongly suggest using it here too!

I've used "erase" in its stead here!

cdsolari reviewed (approved):
I reviewed the code that JP added for the final evaluation form, and it checks out. I re-reviewed the final output files, and they still look good. This is complete!
